pkg/component/etcd/serviceaccount/values_helper.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceaccount
+
+import (
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+// GenerateValues generates `serviceaccount.Values` for the serviceaccount component with the given `etcd` object.
+func GenerateValues(etcd *druidv1alpha1.Etcd, disableEtcdServiceAccountAutomount bool) *Values {
+	return &Values{
+		Name:      etcd.GetServiceAccountName(),
+		Namespace: etcd.Namespace,
+		Labels:    utils.MergeStringMaps(getDefaultLabels(etcd.Name), etcd.Spec.Labels),
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				APIVersion:         druidv1alpha1.GroupVersion.String(),
+				Kind:               "Etcd",
+				Name:               etcd.Name,
+				UID:                etcd.UID,
+				Controller:         pointer.Bool(true),
+				BlockOwnerDeletion: pointer.Bool(true),
+			},
+		},
+		DisableAutomount: disableEtcdServiceAccountAutomount,
+	}
+}
+
+func getDefaultLabels(etcdName string) map[string]string {
```

Comment:
I see that these 2 labels are common for most of the resources that are created for etcd. E.g. `PDB`, `SA`, `ConfigMap` etc. Would it make sense to have a method directly on `Etcd` in `types_etcd.go` called `GetDefaultLabels`
```go
func (e *Etcd) GetDefaultLabels() map[string]string {
   	return map[string]string{
		"name":     "etcd",
		"instance": etcdName,
	}
}
```
Have a constant for the keys somewhere as well.

Reply:
Fixed and raised PR #559 for other components